chore: rename autonomi crate and binary

jacderida opened the pull request:
The `autonomi_cli` binary is renamed to `autonomi` because the `_cli` suffix is not necessary.

The crate is named `autonomi-cli` rather than `autonomi_cli` because we agreed we prefer underscores.

oetyng commented on sn_api/src/app/test_helpers.rs:
```diff
@@ -86,44 +86,37 @@ pub async fn get_next_bearer_dbc() -> Result<(Dbc, Token)> {
 
     let safe = new_safe_instance().await?;
+
+    // we need to get the fees into the outputs first
```
old comment, remove

oetyng commented on sn_api/src/app/wallet.rs:
```diff
@@ -395,17 +429,47 @@ impl Safe {
+    /// Used when reissuing from Wallet.
     #[cfg(feature = "data-network")]
-    fn select_inputs(
+    fn get_reissue_components(
```
This fn can probably be removed, as we have feat gates on the other one.

oetyng commented on sn_api/src/app/wallet.rs:
```diff
@@ -453,72 +517,141 @@ impl Safe {
+    }
+
+    /// B.2 Second step in B-flow (payment-network).
+    async fn select_inputs_with_fees(
```
This can be renamed to `select_inputs` when `get_reissue_components` fn is removed.

oetyng commented on sn_api/src/app/wallet.rs:
```diff
@@ -453,72 +517,141 @@ impl Safe {
+            let revealed_bearer = dbc.as_revealed_input_bearer()?;
+            let input_key = revealed_bearer.public_key();
+
+            // ------------ fee part start? ----------------
```
We can feat flag here and `fn select_inputs` can be removed, while we rename `select_inputs_with_fees` to `select_inputs`.

oetyng commented on sn_api/src/app/wallet.rs:
```diff
@@ -608,15 +756,23 @@ impl Safe {
+    /// This will pay transfer fees if not in data-network.
+    /// The input dbcs will be spent on the network, and the resulting
+    /// dbcs (and change dbc if any) are returned.
+    async fn reissue_dbcs_with_fees(
```
Consider renaming this when the reduntant fns above have been squashed (so that we don't call it "with_fees").

oetyng commented on sn_client/src/api/spend_queries.rs:
```diff
@@ -9,28 +9,26 @@
-    pub async fn get_mint_fees(&self, dbc_key: PublicKey) -> Result<BTreeMap<PublicKey, Token>> {
-        let address = SpentbookAddress::new(XorName::from_content(&dbc_key.to_bytes()));
-        let fee_query = DataQuery::Spentbook(SpendQuery::GetFees(address));
+    pub async fn get_mint_fees(&self, dbc_id: PublicKey) -> Result<BTreeMap<NodeId, RequiredFee>> {
```
Rename `get_mint_fees`, we hardly talk of mints.

oetyng commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -207,6 +210,8 @@ mod tests {
                 genesis_dbc.inputs_spent_proofs,
                 genesis_dbc.inputs_spent_transactions,
+                #[cfg(not(feature = "data-network"))]
+                BTreeMap::new(),
```
Fix this.

oetyng commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -250,6 +255,8 @@ mod tests {
                 invalid_spent_proofs,
                 genesis_dbc.inputs_spent_transactions,
+                #[cfg(not(feature = "data-network"))]
+                BTreeMap::new(),
```
Fix this.

oetyng commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -296,6 +303,8 @@ mod tests {
                 genesis_dbc.inputs_spent_proofs,
                 genesis_dbc.inputs_spent_transactions,
+                #[cfg(not(feature = "data-network"))]
+                BTreeMap::new(),
```
Fix this.

oetyng commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -370,6 +379,8 @@ mod tests {
                 genesis_dbc.inputs_spent_proofs.clone(),
                 genesis_dbc.inputs_spent_transactions,
+                #[cfg(not(feature = "data-network"))]
+                BTreeMap::new(),
```
Fix this.

oetyng commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -417,6 +427,8 @@ mod tests {
                 genesis_dbc.inputs_spent_proofs.clone(),
                 genesis_dbc.inputs_spent_transactions,
+                #[cfg(not(feature = "data-network"))]
+                BTreeMap::new(),
```
Fix this.

oetyng commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -533,6 +545,8 @@ mod tests {
                         spent_transactions.clone(),
+                        #[cfg(not(feature = "data-network"))]
+                        BTreeMap::new(),
                     )
```
Fix this.

oetyng commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -533,6 +545,8 @@ mod tests {
                         spent_proofs.clone(),
                         spent_transactions.clone(),
+                        #[cfg(not(feature = "data-network"))]
+                        BTreeMap::new(),
```
Fix this.

oetyng commented on sn_interface/src/types/fees/fee_ciphers.rs:
```diff
@@ -0,0 +1,73 @@
+    derivation_index: bls::Ciphertext,
+}
+
+impl FeeCiphers {
```
Add a couple of units.

oetyng commented on sn_node/src/node/flow_ctrl/tests/mod.rs:
```diff
@@ -716,59 +716,132 @@ async fn spentbook_spend_client_message_should_replicate_to_adults_and_send_ack(
+        let fee_base_pk = context.reward_secret_key.public_key();
+        let derivation_index_cipher = fee_base_pk.encrypt(fee_derived_owner.derivation_index);
+
+        // We must generate the revealed output (i.e. the blinding factor) to be both encrypted, and used in the dbc
```
Old comment. Remove or rephrase.

oetyng commented on sn_node/src/node/flow_ctrl/tests/mod.rs:
```diff
@@ -916,6 +998,8 @@ async fn spentbook_spend_with_updated_network_knowledge_should_update_the_node()
             spent_transactions: new_dbc2.inputs_spent_transactions,
             network_knowledge: Some((proof_chain, sap)),
+            #[cfg(not(feature = "data-network"))]
+            fee_ciphers: BTreeMap::new(),
```
Consider fixing this. It's not necessary since we're not validating that part, and have covered it other tests here.

oetyng commented on sn_node/src/node/messaging/client_msgs.rs:
```diff
@@ -49,9 +55,16 @@ impl MyNode {
+            // We receive this directly from client, as an Elder, since `is_spend` is set to true (that is a very
+            // messy/confusing pattern, to be fixed).
+
+            // This is using the invoice type a bit outside its intended use. The client is asking for the fee to spend
```
Old comment about invoice. Rephrase.

oetyng commented on sn_api/src/app/wallet.rs:
```diff
@@ -453,72 +517,141 @@ impl Safe {
+            let revealed_bearer = dbc.as_revealed_input_bearer()?;
+            let input_key = revealed_bearer.public_key();
+
+            // ------------ fee part start ----------------
```
We can feat flag here and `fn select_inputs` can be removed, while we rename `select_inputs_with_fees` to `select_inputs`.

maqi commented on sn_api/src/app/wallet.rs:
```diff
@@ -680,15 +510,16 @@ impl Safe {
-    pub(super) async fn reissue_dbcs(
+    /// The input dbcs will be spent on the network, and the resulting
+    /// dbcs (and change dbc if any) are returned.
+    /// This will pay transfer fees if not in data-network.
```
The comment for the `reissue_dbcs` function is the same to part of the comment of `fn send_tokens` .
then what's the diff between these two?
The inputs of this function has `outputs` as input, what's is it for?

oetyng replied:
Good point. I will update this to make more sense.
The outputs are really the recipients, as they are more clearly named in the upper `send_tokens` api.
They must be included so we know where to send the tokens. 
Reissue really just means sending from set of dbcs A to recipients B.

oetyng commented on sn_api/src/app/wallet.rs:
```diff
@@ -1237,10 +1062,14 @@ mod tests {
-        tokio::time::sleep(tokio::time::Duration::from_millis(5000)).await;
+        // 2 dbc inputs = 2 fees
+        let fee_1 = get_spend_fee(&safe, &dbc1).await?;
+        let fee_2 = get_spend_fee(&safe, &dbc1).await?;
```
Woops, this should be dbc2. Will fix.

oetyng commented on sn_api/src/app/wallet.rs:
```diff
@@ -1620,17 +1455,22 @@ mod tests {
-        tokio::time::sleep(tokio::time::Duration::from_millis(1000)).await;
+        // 2 dbc inputs = 2 fees
+        let fee_1 = get_spend_fee(&safe, &dbc1).await?;
+        let fee_2 = get_spend_fee(&safe, &dbc1).await?;
```
Same here, should be dbc for fee_2.

oetyng commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -439,27 +333,21 @@ mod tests {
             }) => {
                 let correct_error_str =
-                    format!("SpentbookError(\"There are no amounts for the given public key {random_public_key:?}\"");
+                    format!("{:?}", sn_dbc::Error::MissingAmountForPubkey(dbc_id_2)); // dbc_id_1
```
remove this comment `// dbc_id_1`

oetyng commented on sn_client/src/api/transfers/mod.rs:
```diff
@@ -0,0 +1,456 @@
+) -> Result<ReissueInputs> {
+    // We'll combine one or more input DBCs and reissue:
+    // - one output DBC for the recipient,
+    // - and a second DBC for the change, which will be stored in the source wallet.
```
This comment is not relevant here: `which will be stored in the source wallet.` as we're oblivious of any wallet here (i.e. this api can theoretically be called both from a wallet and without it).

It's an old comment from when this code was in sn_api wallet.rs.

oetyng commented on sn_client/src/api/transfers/mod.rs:
```diff
@@ -0,0 +1,456 @@
+
+            // Fees that were not encrypted to us.
+            let mut invalid_fees = BTreeSet::new();
+            // As the mints encrypt the amount to our public key, we need to decrypt it.
```
> `mints`
This is old terminology. Update this comment.

jacderida committed:
chore: rename autonomi crate and binary

The `autonomi_cli` binary is renamed to `autonomi` because the `_cli` suffix is not necessary.

The crate is named `autonomi-cli` rather than `autonomi_cli` because we agreed we prefer
underscores.
